src/core/packages/chrome/browser-internal/src/ui/project/sidenav_v2/navigation/to_navigation_items.tsx

```diff
@@ -412,6 +415,31 @@ const isRecentlyAccessedDefinition = (
 const filterEmpty = <T,>(arr: Array<T | null | undefined>): T[] =>
   arr.filter((item) => item !== null && item !== undefined) as T[];
 
+function warnAboutDuplicateIcons(logoItem: SideNavLogo, primaryItems: MenuItem[]) {
+  // Collect all items with icons
+  const itemsWithIcons = [logoItem, ...primaryItems]
+    .filter((item) => item.iconType && item.iconType !== FALLBACK_ICON)
+    .map((item) => ({ id: item.id, icon: String(item.iconType) }));
```

Comment:
nit: Maybe use reduce to traverse only once through the list.
```suggestion
  const itemsWithIcons = [logoItem, ...primaryItems].reduce<Array<{ id: string; icon: string }>>(
    (acc, item) => {
      if (item.iconType && item.iconType !== FALLBACK_ICON) {
        acc.push({ id: item.id, icon: String(item.iconType) });
      }
      return acc;
    },
    []
  );
```